### PR TITLE
Added field to the RealmImport spec to replace environment variables within the realm import

### LINKED ISSUE #26470: Add a field to the RealmImportSpec to toggle replacing ENV variable placeholders
### Description

It would be nice if the RealmImport CRD just had a boolean to toggle replacing ENV variables within the realm import as opposed to needing to use the JAVA_OPTS_APPEND field, since that is not very intuitive.

### Discussion

_No response_

### Motivation

The current way of enabling this feature is not intuitive and is going to keep coming up.

### Details

_No response_

### PR DESCRIPTION
**Pull request by shawkins**
Closes #26470
Replaces #26471

### CODE REVIEW

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/PlaceholderSecret.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import io.sundr.builder.annotations.Buildable;
+
+import java.util.Objects;
+
+/**
+ * @author Scott Tustison
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class PlaceholderSecret {
```
Comment (mabartos) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/PlaceholderSecret.java:31]: I can imagine a few different situations here.

1. Support **only secrets** for placeholders - If we want to support only secrets, we might not need the new object `PlaceholderSecret`, but only the `SecretKeySelector`, right? 
2. Support **secrets** and **config maps** for placeholders - If we want to support both of them and have them both in the same object, I'd choose a different name for the class. 

I think the 2nd option is more accurate, so I'd propose only this nitpick related to changing the class name for something like this:
```suggestion
public class Placeholder {
```

WDYT?

**Review by mabartos — COMMENTED** [review]
@stustison @shawkins Looks very good! Just one nitpick.

@vmuzikar @Pepo48 Could you please check it?

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/PlaceholderSecret.java (hunk 1 above)
Comment (shawkins) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/PlaceholderSecret.java:31]: @mabartos seems fine to change the class name.

**Review by mabartos — DISMISSED** [review]
@shawkins Better, thanks! LGTM

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/KeycloakRealmImportSpec.java [hunk 2]
```diff
@@ -35,6 +42,9 @@ public class KeycloakRealmImportSpec {
     @JsonPropertyDescription("Compute Resources required by Keycloak container. If not specified, the value is inherited from the Keycloak CR.")
     private ResourceRequirements resourceRequirements;
 
+    @JsonPropertyDescription("Optionally set to replace ENV variable placeholders in the realm import.")
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/KeycloakRealmImportSpec.java:45]: I would not mention env vars. It's an implementational detail hidden from the user (they don't set any env vars).
```suggestion
    @JsonPropertyDescription("Optionally set to replace placeholders in the realm import.")
```

**Review by vmuzikar — COMMENTED** [review]
@stustison @shawkins Thank you both for the changes! LGTM, I'd just suggest to add some short example to the RealmImport CR docs and maybe a mention to release notes too.

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/KeycloakRealmImportSpec.java (hunk 2 above)
Comment (shawkins) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/realmimport/KeycloakRealmImportSpec.java:45]: The docs https://www.keycloak.org/server/importExport do specifically say the placeholders are resolved from environment variables.

**Comment by shawkins** [comment]
> @stustison @shawkins Thank you both for the changes! LGTM, I'd just suggest to add some short example to the RealmImport CR docs and maybe a mention to release notes too.

Added the docs.

**Review by mabartos — COMMENTED** [review]
Some nitpicks -> but mainly the 'on' vs. 'only'

**Comment by shawkins** [comment]
@mabartos thanks for the review, those suggestions have been added.

**Review by vmuzikar — APPROVED** [review]
@shawkins LGTM, thank you!

**Review by mabartos — APPROVED** [review]
LGTM

**Pull request merged**